File: kalitech-engine/src/main/resources/kalitech/engine/modules/Editor/EditorLines.js
// FILE: resources/kalitech/engine/modules/Editor/EditorLines.js
// Author: Kalitech
"use strict";

const {requireEngineApi, normalizeCfgObject} = require("../helpers/ModuleCommon.js");

/**
 * Editor lines API wrapper.
 * Manages editor-only grid plane helpers.
 */
function create(engine /*, K */) {
    const api = requireEngineApi(engine, "editorLines", "EDITOR_LINES");

    return Object.freeze({
        createGridPlane(cfg) {
            return api.createGridPlane(normalizeCfgObject(cfg));
        },

        destroy(handle) {
            api.destroy(handle);
            return true;
        },

        api
    });
}

create.META = {
    moduleId: "editorLines",
    globalName: "EDITOR_LINES",
    version: "1.0.0",
    description: "Editor lines wrapper for grid plane helpers",
    engineMin: "0.1.0"
};

module.exports = create;

File: kalitech-engine/src/main/resources/kalitech/engine/modules/Editor/EditorLines.lua
local M = {}
local lua_require_result_0 = require("../helpers/ModuleCommon.lua")
requireEngineApi = lua_require_result_0.requireEngineApi
normalizeCfgObject = lua_require_result_0.normalizeCfgObject
create = setmetatable(
    {},
    {__call = function(lua_, self, engine)
        local api = requireEngineApi(_G, engine, "editorLines", "EDITOR_LINES")
        return KObject:freeze({
            createGridPlane = function(self, cfg)
                return api:createGridPlane(normalizeCfgObject(_G, cfg))
            end,
            destroy = function(self, handle)
                api:destroy(handle)
                return true
            end,
            api = api
        })
    end}
)
create.META = {
    moduleId = "editorLines",
    version = "1.0.0",
    description = "Editor lines wrapper for grid plane helpers",
    engineMin = "0.1.0"
}
M = create

return M
